common/txmgr/broadcaster.go

```diff
@@ -312,6 +325,33 @@ func (eb *Broadcaster[CHAIN_ID, HEAD, ADDR, TX_HASH, BLOCK_HASH, SEQ, FEE]) txIn
 	}
 }
 
+// Load the next sequence map using the tx table or on-chain (if not found in tx table)
+func (eb *Broadcaster[CHAIN_ID, HEAD, ADDR, TX_HASH, BLOCK_HASH, SEQ, FEE]) loadNextSequenceMap(addresses []ADDR) (map[ADDR]SEQ, error) {
+	ctx, cancel := eb.chStop.NewCtx()
+	defer cancel()
+
+	nextSequenceMap := make(map[ADDR]SEQ)
+	for _, address := range addresses {
+		// Get the highest sequence from the tx table
+		// Will need to be incremented since this sequence is already used
+		seq, err := eb.txStore.FindHighestSequence(ctx, address, eb.chainID)
+		if err != nil {
+			// Look for nonce on-chain if address not found in TxStore
+			// Returns the nonce that should be used for the next transaction so no need to increment
+			seq, err = eb.client.PendingSequenceAt(ctx, address)
+			if err != nil {
+				return nil, errors.New("failed to retrieve next sequence from on-chain causing failure to load next sequence map on broadcaster startup")
+			}
+
+			nextSequenceMap[address] = seq
+		} else {
+			nextSequenceMap[address] = eb.generateNextSequence(seq)
```

Comment:
If seq is returned as 0 from txStore.FindHighestSequence, which is possible if there were no recent txes for a given address, currently you are setting 1 as the next nonce in this map.
But that would be wrong.
If yo get a 0, then you should get the seq from eb.client.PendingSequenceAt().

Also, can you please add a unit-test for this case?

Reply:
In the case there's an entry in `evm.txes` with nonce 0, I think we'd want to set 1. I'll rework the query with ORDER BY and LIMIT to be able to differentiate between 0 nonce and no rows error. It won't be as efficient but I think it should be negligible for our data set and it's only at start up